### PR TITLE
Date component validation fix

### PR DESCRIPTION
**Pull request by mallenscdhhs**
Fixed an issue where Date field would clear out any entered value after validation. Also cleaned up the utils#getDateString method, some tests.

### CODE REVIEW

File: src/Date.jsx [hunk 1]
```diff
@@ -3,7 +3,6 @@ import React from 'react';
 import _ from 'lodash';
 import { DateTimePicker } from 'react-widgets';
 import ValueChangeMixin from './ValueChangeMixin';
-import utils from './utils';
```
Comment (mallenscdhhs) [inline src/Date.jsx]: wha?

File: src/Date.jsx [hunk 2]
```diff
@@ -59,18 +58,15 @@ let DateField = React.createClass({
    * passed into this.props.format
    */
   componentWillMount() {
-    let value = DateField.getDateValueFromProps(this.props);
+    let value = DateField.getDateValue(this.props.value);
     this.setState({value});
   },
 
-  componentDidMount() {
```
Comment (mallenscdhhs) [inline src/Date.jsx]: not sure what happened here, merging in your stuff got borked... will fix

File: test/unit/date-spec.js [hunk 3]
```diff
@@ -58,4 +58,14 @@ describe('Date input', function() {
...
   });
 
+  describe('#getDateValue', () => {
+    it('will return today\'s date if value is "today"', () => {
+      let result = DateComponent.getDateValue('today').toDateString();
+      let now = new Date().toDateString();
+      expect(result).toBe(now);
+    });
+
+    it('will parse and return a UTC timestamp');
```
Comment (mallenscdhhs) [inline test/unit/date-spec.js]: whoops, forgot about this little guy... give me 5mins

**Comment by kjschulz** [comment]
:+1:

**Pull request merged**